syntax = "proto3";  // 声明protobuf版本

package fixbug;  // 声明一个namespace

// 定义选项, 表示生成service相关的类
option cc_generic_services = true; 

message ResultCode
{
    int32 errcode = 1;
    bytes errmsg = 2;
}

message LoginRequest
{
    bytes name = 1;
    bytes pwd = 2;
    map<int32, string> test = 3;
}

message LoginResponse
{
    ResultCode result = 1;
    bool success = 3;
}

message User
{
    bytes name = 1;
    int32 age = 2;
    enum Sex
    {
        MAN = 0;
        WOMAN = 1;
    }
    Sex sex = 3;
}

message GetFriendListsRequest
{
    uint32 userid = 1;
}

message GetFriendListsResponse
{
    ResultCode result = 1;
    repeated User friend_list = 2; // 定义列表类型
}


// 对rpc请求做实参和返回值的序列化和反序列化
service UserServiceRpc
{
    rpc Login(LoginRequest) returns(LoginResponse);
    rpc GetFriendLists(GetFriendListsRequest) returns(GetFriendListsResponse);
}